core/src/test/scala/com/softwaremill/events/EventMachineTest.scala

```diff
@@ -90,6 +94,25 @@ class EventMachineTest extends FlatSpec with Matchers with SqlSpec with Eventual
       m.asyncEventRunner.stop()
     }
   }
+
+  it should "recover stored events" in {
+    // given
+    var actions = Vector.empty[String]
+
+    val m = createModules(Registry()
+      .registerModelUpdate[Event1] { e => DBIOAction.successful(()).map { r => actions :+= "mu1" + e.data.data; r } }
+      .registerEventListener[Event1] { e => DBIOAction.successful(Nil).map { r => actions :+= "el1"; r } })
+
+    implicit val hc = HandleContext.System
+    val result = m.eventMachine.run(CommandResult.successful((), Event(Event1("x")).forNewAggregate)).futureValue
+
+    // when
+    val failOverFuture = m.eventMachine.failOverFromStoredEventsTo().futureValue
```

Comment:
+1 for:
restore the system as it was at a given time

I have a usecase where this could be valuable in reconstructing a certain ordering of events when an entity of type A moves to an entity of type B.